### PR TITLE
Add support to filter by group

### LINKED ISSUE #10: Feature Request: Sorting users based on roles
This will probably solve the need for having a team page as people always want to know who are the ones handling the community. Since all the user sortings are already in this extension so maybe, sorting users based on roles would also fit here.

### PR DESCRIPTION
**Pull request by askvortsov1**
Fixes #10 

Changelog:
- Added GroupIDGambit, as otherwise we can't support groups with spaces in the name
- Added logic and UI for group filtering (using group ID gambit)

Demo Video:
https://youtu.be/tIkyLAFA8cE

### CODE REVIEW

**Comment by askvortsov1** [comment]
I'm considering expanding this so that stuff like #5 (the filter by suspend especially) could be added; we'd have dividers in between groups of buttons, the first group would be for groups, and others, like suspended status, could be added

File: src/Search/Gambit/GroupIdGambit.php [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+     */
+    protected function conditions(AbstractSearch $search, array $matches, $negate)
+    {
+        if (!$search instanceof UserSearch) {
+            throw new LogicException('This gambit can only be applied on a UserSearch');
+        }
+
+        $groupIds = $this->extractGroupIds($matches);
+
+        // TODO: Use a JOIN instead (and don't forget to remove the findByName() method again)
```
Comment (clarkwinkelmann) [inline src/Search/Gambit/GroupIdGambit.php:50]: Is that TODO relevant?

Comment (askvortsov1) [reply src/Search/Gambit/GroupIdGambit.php:50]: Nope, it's the result of a copy-paste. I agree with this, but what do you think about https://github.com/FriendsOfFlarum/user-directory/issues/5#issuecomment-640102590? Also, should this be integrated into the group gambit (which might be better tbh), or as it's own gambit?

**Comment by clarkwinkelmann** [comment]
@askvortsov1 are you interested in updating this to use the new native group gambit?

Maybe you'll want to create a new PR for that. Feel free to close this one.

**Comment by askvortsov1** [comment]
Updated

**Review by imorland — APPROVED** [review]
Checked out locally, appears to be as intended 👍 

Thanks @askvortsov1

**Pull request merged**